node/src/types/block.rs

```diff
@@ -1111,6 +1113,217 @@ impl Item for BlockHeaderWithMetadata {
         self.block_header.height()
     }
 }
+#[derive(DataSize, Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+/// ID idenfitifying a request for a batch of block headers.
+pub struct BlockHeadersBatchId {
+    pub highest: u64,
+    pub lowest: u64,
+}
+
+impl BlockHeadersBatchId {
+    pub fn new(highest: u64, lowest: u64) -> Self {
+        Self { highest, lowest }
+    }
+    pub fn from_known(lowest_known_block_header: &BlockHeader, max_batch_size: u64) -> Self {
+        let highest = lowest_known_block_header.height().saturating_sub(1);
+        let lowest = lowest_known_block_header
+            .height()
+            .saturating_sub(max_batch_size);
+
+        Self { highest, lowest }
+    }
+
+    /// Return an iterator over block header heights starting from highest (inclusive) to lowest
+    /// (inclusive).
+    pub fn iter(&self) -> impl Iterator<Item = u64> {
+        (self.lowest..=self.highest).rev()
+    }
+
+    /// Returns a length of the batch.
```

Comment:
```suggestion
    /// Returns the length of the batch.
```

Reply:
Done in [`60ac7e8` (#3020)](https://github.com/casper-network/casper-node/pull/3020/commits/60ac7e8598c652d5e8d043c149df8319bdd03027)